### PR TITLE
Fixed bug in __eq__ for Factor

### PR DESCRIPTION
**Pull request by khalibartan**
Ping @ankurankan 
Should I add test for it.?

### CODE REVIEW

**Comment by ankurankan** [comment]
@khalibartan Yes please add tests for all the changes that you make.

**Comment by ankurankan** [comment]
@khalibartan Can you please add the tests for this ?

**Comment by khalibartan** [comment]
@ankurankan Oh ! I forgot. Just sending in a patch.

File: pgmpy/tests/test_factors/test_Factor.py [hunk 1]
```diff
@@ -264,7 +264,8 @@ def test_eq1(self):
                                                       4, 5, 15, 16, 17, 6, 7,
                                                       8, 18, 19, 20, 9, 10, 11,
                                                       21, 22, 23])
-        self.assertTrue(phi1, phi2)
+        self.assertTrue(phi1 == phi2)
+        self.assertEqual(phi2.variables,['x2', 'x1', 'x3'] )
```
Comment (khalibartan) [inline pgmpy/tests/test_factors/test_Factor.py:268]: @ankurankan This test is enough because, earlier it would have returned `['x1', 'x2', 'x3']`

**Comment by ankurankan** [comment]
@khalibartan Tests are failing. Check please.

**Comment by khalibartan** [comment]
Yes I'm checking into it.

**Comment by khalibartan** [comment]
@ankurankan Fixed the issue.

File: pgmpy/factors/JointProbabilityDistribution.py [hunk 2]
```diff
@@ -185,12 +185,12 @@ def check_independence(self, event1, event2, event3=None, condition_random_varia
                 # Using the definition of conditional independence
                 # If P(X,Y|Z) = P(X|Z)*P(Y|Z)
                 # This can be expanded to P(X,Y,Z)*P(Z) == P(X,Z)*P(Y,Z)
-                jpd_z = JPD.marginal_distribution(event3, inplace=False)
+                phi_z = JPD.marginal_distribution(event3, inplace=False).to_factor()
                 for variable_pair in itertools.product(event1, event2):
-                    jpd_xyz = JPD.marginal_distribution(event3 + list(variable_pair), inplace=False)
-                    jpd_xz = JPD.marginal_distribution(event3 + [variable_pair[0]], inplace=False)
-                    jpd_yz = JPD.marginal_distribution(event3 + [variable_pair[1]], inplace=False)
-                    if jpd_xyz * jpd_z != jpd_xz * jpd_yz:
```
Comment (khalibartan) [inline pgmpy/factors/JointProbabilityDistribution.py:193]: not necessarily a JPD, didn't pay attention to it. Sorry for the inconvenience.

**Pull request merged**